Add docs about EEM and new Services inventory (#4062)
elastic/observability-docs: docs/en/serverless/apm/new-experience-services.mdx

Reviewer: > To turn off the new experience and restore the "classic" view:

@dedemorton - Something I didn't call out is that opting in and out is done at a browser level. This means that users can try it and switch back and forth **without** affecting other users. I think this could be worth mentioning in case admins/users get nervous about affecting other users' experiences (i.e. this isn't a feature flag which affects all users - it's **per browser**).

In short:

- Once you 'try it', it'll turn on EEM and also switch the UI to the new experience for the browser you are in
- If you 'return to classic', EEM will stay on* but the UI will revert to the existing experience for this browser
- Users can flip back and forth between the 'new' and 'classic' experience whenever they want - it **won't** affect other users
 
_*EEM is going to be default functionality...the only reason we need to mention it in the UI is because we've not figured out how to get around some permissions issues. In short, EEM is going to be running in the background for all users as soon as we can do it._

Author: @roshan-elastic I'm assuming the browser-level setting that enables/disables the new experience is persisted between browser restarts and tabs/windows, correct? (TBH I'm not sure how browser settings like this are scoped.)

Author: I've updated the docs with the assumption that these settings are persisted across browser sessions. Just let me know if the info needs to be changed.

Reviewer: > @roshan-elastic I'm assuming the browser-level setting that enables/disables the new experience is persisted between browser restarts and tabs/windows, correct? (TBH I'm not sure how browser settings like this are scoped.)

That's right @dedemorton. It'll stay with you in the current browser unless you delete your localStorage files (i.e. clear your browser settings etc).

With this technical preview,
Elastic expands the **Services** inventory to include services found in logs.
This means you can monitor the services you care about even if you only have logs
and aren't ready to instrument your services.

The new **Services** inventory shows services instrumented with APM or OpenTelemetry,
plus any services declared using `service.name` in your logs:

<DocImage url="../images/entity-centric-services-inventory.png" alt="Services inventory that shows APM data and log data"/>

For instrumented services that also have custom logs, the APM and log data are displayed in the same view.
For services that only have log data,
the log data is displayed, and you have the option of adding APM later.

<DocCallOut title="Note">
   The new **Services** inventory requires the Elastic Entity Model (EEM). To learn more, refer to <DocLink slug="/serverless/observability/elastic-entity-model" />.
</DocCallOut>

## Turn on the new experience

The new services experience is turned off by default. To turn it on:

1. In your ((observability)) project, go to **Applications** → **Services**.
1. Click **Try out Elastic's new experience!**

> If you have **Admin** privileges, the Elastic Entity Model will be turned on automatically.

Reviewer 2: ```suggestion
If you have **Admin** privileges, the EEM will be turned on automatically.
```
Not sure if you want to use it or not, but you've called out EEM in the previous note.

Otherwise, you need to ask someone with **Admin** privileges to turn on this feature under **Stack Management**.

To turn off the new experience, click **Restore classic view**.
You can switch back and forth between the new and classic views whenever you want.

> <DocCallOut title="Does opting into the new experience affect other users?">

Reviewer 2: ```suggestion
<DocCallOut title="Does turning on the new experience affect other users?">
```
Just this one last "opting into".

Author: sigh...thanks. :-)

> <DocCallOut title="Does opting into the new experience affect other users?">

Reviewer 2: "opting into" feels a little different than the "turn on" language we were using previously in the section. I noticed the EEM doc also says opting in, not a big deal most likely, but might be better to stick to one unless they mean different things?

Author: good point!

   No. The settings are applied at the browser level,

>    which means that other users need to opt in if they want to use the new experience.

Author: ```suggestion
   which means that other users need to turn on the new experience if they want to use it.
```

</DocCallOut>

## Add services to the inventory

To add services to the inventory, use one of the following approaches:

* Instrument your services with APM or OpenTelemetry.
To learn how, refer to <DocLink slug="/serverless/observability/apm-send-data-to-elastic" />.

> * Declare `service.name` in your logs.

Author: After Mike's PR is merged, need to add "To learn how, refer to <DocLink slug="/serverless/observability/add-logs-service-name" />."

> * Declare `service.name` in your logs.

Author: ```suggestion
* Declare `service.name` in your logs. To learn how, refer to <DocLink slug="/serverless/observability/add-logs-service-name" />
```

For instrumented services that also have custom logs, the APM and log data are displayed in the same view.
For services that only have log data,
the log data is displayed, and you have the option of adding APM later.

<DocCallOut title="Note">
   The new **Services** inventory requires the Elastic Entity Model (EEM). To learn more, refer to <DocLink slug="/serverless/observability/elastic-entity-model" />.
</DocCallOut>

## Turn on the new experience

The new Services inventory is turned off by default. To turn it on:

1. Go to **Applications** → **Services**.
1. Click **Try out Elastic's new experience!**

If you have **Admin** privileges, the Elastic Entity Model will be turned on automatically.
Otherwise, you need to ask someone with **Admin** privileges to turn on this feature under **Stack Management**.

## Add services to the inventory

To add services to the inventory, use one of the following approaches:

* Instrument your services with APM or OpenTelemetry.
To learn how, refer to <DocLink slug="/serverless/observability/apm-send-data-to-elastic" />.
* Declare `service.name` in your logs.

> To learn how, refer to LINK TBD. {/* TODO: Add link to topic that Mike is working on.*/}

Author: Looks like https://github.com/elastic/observability-docs/pull/4054 defines the slug as /serverless/observability/add-logs-service-name.

</DocCallOut>

## Turn on the new experience

The new services experience is turned off by default. To turn it on:

1. In your ((observability)) project, go to **Applications** → **Services**.
1. Click **Try out Elastic's new experience!**

If you have **Admin** privileges, the Elastic Entity Model will be turned on automatically.
Otherwise, you need to ask someone with **Admin** privileges to turn on this feature under **Stack Management**.

To turn off the new experience, click **Restore classic view**.
You can switch back and forth between the new and classic views whenever you want.

<DocCallOut title="Does opting into the new experience affect other users?">
   No. The settings are applied at the browser level,
   which means that other users need to opt in if they want to use the new experience.
</DocCallOut>

## Add services to the inventory

To add services to the inventory, use one of the following approaches:

* Instrument your services with APM or OpenTelemetry.
To learn how, refer to <DocLink slug="/serverless/observability/apm-send-data-to-elastic" />.

> * Declare `service.name` in your logs.
> {/* TODO: Activate link after https://github.com/elastic/observability-docs/pull/4054 is merged To learn how, refer to <DocLink slug="/serverless/observability/add-logs-service-name" />.*/}

Author: ```suggestion
* Declare `service.name` in your logs.
```
